### ISSUE TITLE
How does Axios handles mTLS requests?

### ISSUE THREAD

**Issue by alannaidon** [axios/axios#6474]
### Describe the issue

Hi!
Let's say I make a request with mTLS certificate loaded.  
```
import * as fs from 'node:fs';
import * as https from 'node:https';

let axiosConfig = {
  headers: { 'Content-Type': 'application/json' };
}

const httpsAgent = new https.Agent({
  key: fs.readFileSync(path.join(__dirname, 'client.key')),
  cert: fs.readFileSync(path.join(__dirname, 'client.crt')),
  ca: fs.readFileSync(path.join(__dirname, 'ca.crt')),
  rejectUnauthorized: true,
});

axiosConfig = {
  ...axiosConfig,
  httpsAgent,
};
const response = await axios.post(webhookUrl, eventDto, axiosConfig);
```  

Usually, a server that uses mTLS will request for the certificate. When the request is made to a server without mTLS, so the server does NOT request the certificate. What Happens? What is Axios default behaviour? 

Is there a way to **force** it? I mean, If I load the mTLS certificate, but the server doesn't use it I don't want the request to complete. Is it possible?

**Comment by Lanna253** [comment]
Its a Jason module I spell wrong I found in Babel it's a dependency

**Comment by jasonsaayman** [comment]
No. Neither Axios nor Node’s https.Agent provide a way to require that the server request a client certificate or to fail the connection if it doesn’t. This behaviour is governed by the TLS handshake, and Node.js doesn’t expose a control to change it.

Axios itself has no special mTLS logic; all behaviour depends on Node.js’s https.Agent and the TLS protocol. The client cannot force the server to request or require a certificate; that must be enforced on the server side or checked via custom logic on the client.
